src/App.js

```diff
@@ -213,176 +258,242 @@ class Board extends React.Component {
+    positionAndValue.push(board[randomPositionOfBoard]);
 
     //Set value of the random picked position to zero
-    valueBoard[randomPositionOfBoard] = 0;
-    
+    board[randomPositionOfBoard] = 0;
+
     //Tell which position got their number removed and what number
     return positionAndValue;
   }
 
-  //Set the value of a position given by backup
-  resetValueBackToOriginal(undo, valueBoard) {
-    //Reset the position, given from removeRandomNumber method, of the board's value
-    valueBoard[undo[0]] = undo[1];
-  }
-
-  getPossibleNumberInEmptySquares(valueBoard, backup) {
+  getPossibleNumberInEmptySquares(board, backup) {
     let positionAndAllPossibleNumbers = [];
     let oneSetOfPossibleNumbers = [];
-    for(let i = 0; i < backup.length; i++) {
-      for(let j = 1; j < 10; j++) {
-        if( this.isPossibleRow(j, this.returnRow(backup[i][0]), valueBoard) &&
-            this.isPossibleCol(j, this.returnCol(backup[i][0]), valueBoard) &&
-            this.isPossibleBlock(j, this.returnBlock(backup[i][0]), valueBoard)
+    for (let i = 0; i < backup.length; i++) {
+      for (let j = 1; j < 10; j++) {
+
+        if (this.isPossibleRow(j, this.returnRow(backup[i]), board) &&
+          this.isPossibleCol(j, this.returnCol(backup[i]), board) &&
+          this.isPossibleBlock(j, this.returnBlock(backup[i]), board)
         ) {
           oneSetOfPossibleNumbers.push(j);
         }
       }
-      positionAndAllPossibleNumbers.push([backup[i][0], oneSetOfPossibleNumbers]);
+
+      positionAndAllPossibleNumbers.push([backup[i], oneSetOfPossibleNumbers]);
+
       oneSetOfPossibleNumbers = [];
     }
     return positionAndAllPossibleNumbers;
   }
 
-  getPossibleNumberInThisSquares(valueBoard, index) {
+  getPossibleNumberInThisSquares(board, index) {
     let allPossibleNumbers = [];
-    for(let j = 1; j < 10; j++) {
-      if( this.isPossibleRow(j, this.returnRow(index), valueBoard) &&
-          this.isPossibleCol(j, this.returnCol(index), valueBoard) &&
-          this.isPossibleBlock(j, this.returnBlock(index), valueBoard)
+    for (let j = 1; j < 10; j++) {
+      if (this.isPossibleRow(j, this.returnRow(index), board) &&
+        this.isPossibleCol(j, this.returnCol(index), board) &&
+        this.isPossibleBlock(j, this.returnBlock(index), board)
       ) {
         allPossibleNumbers.push(j);
       }
     }
     return allPossibleNumbers;
   }
+  
+  /**
+   * This method returns an array with the location (indicies) of empty squares
+   * (not used, but was originally going to use for solver)
+   * @param {*} valueBoard, 1D array representing gameboard
+   */
+  getEmptySquares(valueBoard) {
+    const emptySquaresList = [];
+
+    for (let index = 0; index < valueBoard.length; index++) {
+      if (valueBoard[index] == undefined) { // if the cell is empty, is it undefined or 0?
+        emptySquaresList.push(index);
+      }
+    }
+    return emptySquaresList;
+  }
```

Comment:
not used, but I was going to originally use this helper method for using the other way of implementing DFS

Reply:
We can leave it there. Maybe one day it'll prove to be useful.